homeassistant/components/climate/xiaomi_miio.py

```diff
@@ -0,0 +1,624 @@
+    def temp(self):
+        """Get temp from the climate."""
+        return int(self.data[1][6:8], 16)
+
+    @property
+    def operation(self):
+        """Get operation from the climate."""
+        if self.data[1][2:3] == '0':
+            return 'off'
+        else:
+            if self.data[1][3:4] == '0':
+                return 'heat'
+            elif self.data[1][3:4] == '1':
+                return 'cool'
+            return 'auto'
+
+
+class MiAcPartner(ClimateDevice):
+    """Representation of a demo climate device."""
+
+    def __init__(self, hass, name, target_humidity,
+                 away, hold, current_fan_mode, current_humidity,
+                 current_swing_mode, current_operation, aux,
+                 target_temp_high, target_temp_low, host,
+                 token, sensor_entity_id, sync, customize):
+
+        """Initialize the climate device."""
+        self.hass = hass
+        self._name = name if name else DEFAULT_NAME
+        self._target_humidity = target_humidity
+        self._away = away
+        self._hold = hold
+        self.host = host
+        self.token = token
+        self.sync = sync
+        self._customize = customize
+
+        self._climate = None
+        self._state = None
+        self._state = self.climate_get_state()
+
+        self._target_temperature = self._state.temp
+        self._current_operation = self._state.operation
+
+        self._current_humidity = current_humidity
+        self._aux = aux
+        self._operation_list = ['heat', 'cool', 'auto', 'off']
+
+        if self._customize and ('fan' in self._customize):
+            self._customize_fan_list = list(self._customize['fan'])
+            self._fan_list = self._customize_fan_list
+            self._current_fan_mode = 'idle'
+        else:
+            self._fan_list = ['low', 'medium', 'high', 'auto']
+            self._current_fan_mode = self._state.wind_force
+
+        if self._customize and ('swing' in self._customize):
+            self._customize_swing_list = list(self._customize['swing'])
+            self._swing_list = self._customize_swing_list
+            self._current_swing_mode = 'idle'
+        else:
+            self._swing_list = ['on', 'off']
+            self._current_swing_mode = self._state.sweep
+        self._target_temperature_high = target_temp_high
+        self._target_temperature_low = target_temp_low
+        self._max_temp = target_temp_high + 1
+        self._min_temp = target_temp_low - 1
+        self._target_temp_step = DEFAULT_STEP
+
+        self._unit_of_measurement = TEMP_CELSIUS
+        self._current_temperature = None
+        self._sensor_entity_id = sensor_entity_id
+
+        if sensor_entity_id:
+            async_track_state_change(
+                hass, sensor_entity_id, self._async_sensor_changed)
+            sensor_state = hass.states.get(sensor_entity_id)
+            if sensor_state:
+                self._async_update_temp(sensor_state)
+        if sync:
```

Comment:
This looks like you implement polling yourself. That's what entities have a `should_poll` property for. If that returns `True`, home assistant will poll your `update` method in regular intervals. Please don't implement polling yourself.